app/src/proxy.ts

```diff
@@ -0,0 +1,71 @@
+import child_process from "node:child_process";
+
+export type ProxyRequest = {
+  method: string;
+  path_query: string;
+  stream: false;
+  body: string;
+  headers: object;
+};
+
+export type ProxyResponse = {
+  stdout: string;
+  stderr: string;
+  code: number;
+};
+
+const DEFAULT_PROXY_VM_NAME = "sd-proxy";
+
+export async function proxy(request: ProxyRequest): Promise<ProxyResponse> {
+  let proxyCommand = "";
+  if (import.meta.env.MODE != "development") {
+    const proxyVmName = import.meta.env.VITE_PROXY_VM_NAME
+      ? import.meta.env.VITE_PROXY_VM_NAME
+      : DEFAULT_PROXY_VM_NAME;
+    proxyCommand = `/usr/lib/qubes/qrexec-client-vm ${proxyVmName} securedrop.Proxy`;
```

Comment:
Can we pass the arguments (` ${proxyVmName} securedrop.Proxy`) as arguments to child_process.spawn instead of having them be part of the command string?